File: python/ccxt/static_dependencies/starknet/net/account/account_deployment_result.py
from dataclasses import dataclass

from contract import SentTransaction
from utils.sync import add_sync_methods


@add_sync_methods
@dataclass(frozen=True)
class AccountDeploymentResult(SentTransaction):
    """
    Result of the :meth:`Account.deploy_account <starknet_py.net.account.account.Account.deploy_account>` method.
    """

    account: "Account" = None  # pyright: ignore
    """Account instance created during the deployment."""

    def __post_init__(self):
        if self.account is None:
            raise ValueError(
                "Parameter account cannot be None in AccountDeploymentResult."
            )

File: cs/ccxt/static/StarkSharp/StarkSharp.Base/StarkSharp.Net/Account/account_deployment_result.cs
using StarkSharp.Base.Net.Transaction;
using System;

namespace StarkSharp.Accounts
{
    public class AccountDeploymentResult : BaseTransaction
    {
        public Account Account { get; }
        public AccountDeploymentResult(Account account)
            : base()
        {
            Account = account ?? throw new ArgumentNullException(nameof(account), "Parameter account cannot be None in AccountDeploymentResult.");
        }
    }
}
